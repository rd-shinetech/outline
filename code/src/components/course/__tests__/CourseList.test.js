import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CourseList from '../CourseList';


describe('CourseList.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            courses: [
                { id: 1, title: 'Java Clean Code' },
                { id: 2, title: 'Java The Good Pards' },                
            ],
            handleRowSelect: jest.fn()            
        };
        
        const wrapper = shallow(<CourseList {...props}/>);

        expect(wrapper).toHaveLength(1);
        expect(wrapper.find('BootstrapTable')).toHaveLength(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
