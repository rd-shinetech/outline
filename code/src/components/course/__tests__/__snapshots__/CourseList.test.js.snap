exports[`CourseList.test.js renders without crashing 1`] = `
<BootstrapTable
  bodyContainerClass={null}
  bordered={false}
  cellEdit={
    Object {
      "afterSaveCell": undefined,
      "beforeSaveCell": undefined,
      "blurToSave": false,
      "mode": "none",
      "nonEditableRows": undefined,
    }
  }
  columnFilter={false}
  condensed={true}
  containerClass={null}
  csvFileName="spreadsheet.csv"
  data={
    Array [
      Object {
        "id": 1,
        "title": "Java Clean Code",
      },
      Object {
        "id": 2,
        "title": "Java The Good Pards",
      },
    ]
  }
  deleteRow={false}
  expandColumnOptions={
    Object {
      "expandColumnBeforeSelectColumn": true,
      "expandColumnComponent": undefined,
      "expandColumnVisible": false,
    }
  }
  exportCSV={false}
  fetchInfo={
    Object {
      "dataTotalSize": 0,
    }
  }
  headerContainerClass={null}
  height="100%"
  hover={true}
  ignoreSinglePage={false}
  insertRow={false}
  keyBoardNav={false}
  multiColumnSearch={false}
  multiColumnSort={1}
  options={
    Object {
      "noDataText": "No data",
      "sortIndicator": true,
    }
  }
  pagination={false}
  printable={false}
  search={false}
  selectRow={
    Object {
      "bgColor": "#c1f291",
      "clickToSelect": true,
      "hideSelectColumn": true,
      "mode": "radio",
      "onSelect": [Function],
    }
  }
  striped={true}
  tableBodyClass={null}
  tableContainerClass={null}
  tableHeaderClass={null}
  trClassName="">
  <TableHeaderColumn
    className=""
    columnClassName=""
    columnTitle={false}
    dataAlign="left"
    dataField="id"
    dataSort={false}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={true}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={true}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Id
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="title"
    dataFormat={[Function]}
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filter={
      Object {
        "delay": 0,
        "type": "TextFilter",
      }
    }
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Title
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="length"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Length
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="category"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filter={
      Object {
        "delay": 0,
        "type": "TextFilter",
      }
    }
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Category
  </TableHeaderColumn>
  <TableHeaderColumn
    caretRender={[Function]}
    className=""
    columnClassName=""
    columnTitle={true}
    dataAlign="left"
    dataField="authorId"
    dataSort={true}
    defaultASC={false}
    editColumnClassName=""
    editable={true}
    expandable={true}
    filter={
      Object {
        "delay": 0,
        "type": "TextFilter",
      }
    }
    filterFormatted={false}
    headerTitle={true}
    hidden={false}
    hiddenOnInsert={false}
    invalidEditColumnClassName=""
    isKey={false}
    keyValidator={false}
    searchable={true}
    sortIndicator={true}
    width={null}>
    Author
  </TableHeaderColumn>
</BootstrapTable>
`;
